ApiUsage/ArmBased/Program.cs

```diff
@@ -27,76 +28,106 @@ public static async Task Main(string[] args)
 
             // Get account details
             var account = await videoIndexerResourceProviderClient.GetAccount();
-            var accountId = account.Properties.Id;
             var accountLocation = account.Location;
-            Console.WriteLine($"account id: {accountId}");
-            Console.WriteLine($"account location: {accountLocation}");
+            var accountId = account.Properties.Id;
 
             // Get account level access token for Azure Video Analyzer for Media 
-            var accessTokenRequest = new AccessTokenRequest
-            {
-                PermissionType = AccessTokenPermission.Contributor,
-                Scope = ArmAccessTokenScope.Account
-            };
+            var accountAccessToken = await videoIndexerResourceProviderClient.GetAccessToken(ArmAccessTokenPermission.Contributor, ArmAccessTokenScope.Account);
 
-            var accessToken = await videoIndexerResourceProviderClient.GetAccessToken(accessTokenRequest);
             var apiUrl = "https://api.videoindexer.ai";
             System.Net.ServicePointManager.SecurityProtocol = System.Net.ServicePointManager.SecurityProtocol | System.Net.SecurityProtocolType.Tls12;
 
-
             // Create the http client
-            var handler = new HttpClientHandler();
-            handler.AllowAutoRedirect = false;
+            var handler = new HttpClientHandler
+            {
+                AllowAutoRedirect = false
+            };
             var client = new HttpClient(handler);
 
             // Upload a video
+            var videoId = await UploadVideo(accountId, accountLocation, accountAccessToken, apiUrl, client);
+
+            // Wait for the video index to finish
+            await WaitForIndex(accountId, accountLocation, accountAccessToken, apiUrl, client, videoId);
+
+            // Get video level access token for Azure Video Analyzer for Media 
+            var videoAccessToken = await videoIndexerResourceProviderClient.GetAccessToken(ArmAccessTokenPermission.Contributor, ArmAccessTokenScope.Video, videoId);
```

Comment:
here we can use CW ("Getting Access token for ...." )

Reply:
It's in the get access token function :)